packages/analytics/amplify_analytics_pinpoint_dart/lib/src/exception/pinpoint_analytics_exception.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+AnalyticsException _fromBadRequestException(
+  BadRequestException exception,
+) {
+  return AnalyticsException(
```

Comment:
Following the pattern in Auth/Storage, I think we should make `AnalyticsException` abstract to force sub types to be thrown. This would then become UnknownException. That can be a follow up PR though if you would prefer.

```suggestion
  return UnknownException(
```